### PR TITLE
CI: add scheduled Docker builds workflow

### LINKED ISSUE #2026: Docker nightly builds in GitHub Actions
## Why
We want to detect when the Docker image builds are failing. They are not currently tested in the CI because they are too long to run on each PR.

## What

Implement scheduled CI runs for Docker builds with the available [cron scheduler](https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/events-that-trigger-workflows#schedule) of GitHub Actions.

## How

- [ ] Create new scheduled workflow to build all the Docker images that are not built in the CI (not suffixed by `.ci`)
- [ ] Schedule the run every night at 2 am GMT

### PR DESCRIPTION
**Pull request by dlachaume**
## Content

This PR includes the introduction of a scheduled Docker builds workflow, set to run daily at 2:00 AM UTC, with the option to trigger it manually.

## Pre-submit checklist

- Branch
  - [ ] Tests are provided (if possible)
  - [ ] Crates versions are updated (if relevant)
  - [ ] CHANGELOG file is updated (if relevant)
  - [x] Commit sequence broadly makes sense
  - [x] Key commits have useful messages
- PR
  - [x] No clippy warnings in the CI
  - [x] Self-reviewed the diff
  - [x] Useful pull request description
  - [x] Reviewer requested
- Documentation
  - [ ] Update README file (if relevant)
  - [ ] Update documentation website (if relevant)
  - [ ] Add dev blog post (if relevant)

## Issue(s)

Closes #2026

### CODE REVIEW

**Review by sfauvel — APPROVED** [review]
LGTM

**Review by jpraynaud — APPROVED** [review]
LGTM :+1:

File: .github/workflows/scheduled-docker-builds.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+          [
+            mithril-aggregator,
+            mithril-client-cli,
+            mithril-signer,
+            mithril-relay,
+          ]
+
+        include:
+          - project: mithril-client-cli
+            package: mithril-client
```
Comment (Alenar) [inline .github/workflows/scheduled-docker-builds.yml:24]: What is the utility of those lines ? There's no `matrix.package` usage below.

**Review by Alenar — APPROVED** [review]
LGTM

File: .github/workflows/scheduled-docker-builds.yml (hunk 1 above)
Comment (dlachaume) [reply .github/workflows/scheduled-docker-builds.yml:24]: Indeed, I'll remove this block. It's a leftover from the CI step I used as a reference to create the workflow.

File: .github/workflows/scheduled-docker-builds.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
+name: Scheduled Docker Builds
```
Comment (jpraynaud) [inline .github/workflows/scheduled-docker-builds.yml:1]:
```suggestion
name: Nightly Docker Builds
```

Comment (dlachaume) [reply .github/workflows/scheduled-docker-builds.yml]: Done in the last commit.

**Pull request merged**